src/ORM/FieldType/DBText.php

```diff
@@ -130,10 +139,12 @@ public function Summary($maxWords = 50, $add = false)
             $add = $this->defaultEllipsis();
         }
 
+        $summarySentenceSeparators = implode($this->config()->summarySentenceSeparators);
+
         // Split on sentences (don't remove period)
         $sentences = array_filter(array_map(function ($str) {
             return trim($str);
-        }, preg_split('@(?<=\.)@', $value) ?: []));
+        }, preg_split('@(?<=['. $summarySentenceSeparators .'])@', $value) ?: []));
```

Comment:
Do we need to `preg_quote` something here? What if devs want to split in `@`, what would it do at the current state?

Reply:
agree there is a need to make sure that the strings are escaped

Reply:
In its initial state `preg_split` spits out a "[Warning] Unknown modifier" with characters like `@\\`. If `preg_quote` is used some characters 'll be escaped but just some and php.net even says: Note that preg_quote() is not meant to be applied to the $replacement string(s) of preg_replace() etc.

I'm no regex expert but now have something like bellow working. This way, it's possible to separate with @ or even \ but just single characters can be used as separators which should be OK.
`preg_split('@(?<=[(\.|\;|\:|\?|\!)])@', $value);`

What do you think @dhensby @michalkleiner?

Reply:
> Note that preg_quote() is not meant to be applied to the $replacement string(s) of preg_replace() etc.

That's fine, we aren't using it on the `$replacement` strings because that is the second parameter to `preg_replace`, we're concerned about escaping input in the *pattern*, so `preg_quote` is appropriate.

The correct solution should be:

```php
preg_split('@(?<=['. preg_quote($summarySentenceSeparators, '@') .'])@', $value)
```